mne/epochs.py

```diff
@@ -1324,71 +1417,43 @@ def save(self, fname):
-        start_block(fid, FIFF.FIFFB_EPOCHS)
+            .. versionadded:: 0.10.0
 
-        # write events out after getting data to ensure bad events are dropped
-        data = self.get_data()
-        start_block(fid, FIFF.FIFFB_MNE_EVENTS)
-        write_int(fid, FIFF.FIFF_MNE_EVENT_LIST, self.events.T)
-        mapping_ = ';'.join([k + ':' + str(v) for k, v in
-                             self.event_id.items()])
-        write_string(fid, FIFF.FIFF_DESCRIPTION, mapping_)
-        end_block(fid, FIFF.FIFFB_MNE_EVENTS)
-
-        # First and last sample
-        first = int(self.times[0] * self.info['sfreq'])
-        last = first + len(self.times) - 1
-        write_int(fid, FIFF.FIFF_FIRST_SAMPLE, first)
-        write_int(fid, FIFF.FIFF_LAST_SAMPLE, last)
-
-        # save baseline
-        if self.baseline is not None:
-            bmin, bmax = self.baseline
-            bmin = self.times[0] if bmin is None else bmin
-            bmax = self.times[-1] if bmax is None else bmax
-            write_float(fid, FIFF.FIFF_MNE_BASELINE_MIN, bmin)
-            write_float(fid, FIFF.FIFF_MNE_BASELINE_MAX, bmax)
-
-        # The epochs itself
-        decal = np.empty(self.info['nchan'])
-        for k in range(self.info['nchan']):
-            decal[k] = 1.0 / (self.info['chs'][k]['cal'] *
-                              self.info['chs'][k].get('scale', 1.0))
-
-        data *= decal[np.newaxis, :, np.newaxis]
-
-        write_float_matrix(fid, FIFF.FIFF_EPOCH, data)
-
-        # undo modifications to data
-        data /= decal[np.newaxis, :, np.newaxis]
+        Notes
+        -----
+        Bad epochs will be dropped before saving the epochs to disk.
+        """
+        check_fname(fname, 'epochs', ('-epo.fif', '-epo.fif.gz'))
 
-        write_string(fid, FIFF.FIFFB_MNE_EPOCHS_DROP_LOG,
-                     json.dumps(self.drop_log))
+        if isinstance(split_size, string_types):
+            exp = dict(MB=20, GB=30).get(split_size[-2:], None)
+            if exp is None:
+                raise ValueError('split_size has to end with either'
+                                 '"MB" or "GB"')
+            split_size = int(float(split_size[:-2]) * 2 ** exp)
 
-        write_int(fid, FIFF.FIFFB_MNE_EPOCHS_SELECTION,
-                  self.selection)
+        if split_size > 2147483648:
+            raise ValueError('split_size cannot be larger than 2GB')
 
-        end_block(fid, FIFF.FIFFB_EPOCHS)
+        # to know the length accurately. The get_data() call would drop
+        # bad epochs anyway
+        self.drop_bad_epochs()
+        total_size = self[0].get_data().nbytes * len(self)
+        n_parts = int(np.ceil(total_size / float(split_size)))
+        epoch_idxs = np.array_split(range(len(self)), n_parts)
 
-        end_block(fid, FIFF.FIFFB_PROCESSED_DATA)
-        end_block(fid, FIFF.FIFFB_MEAS)
-        end_file(fid)
+        for part_idx, epoch_idx in enumerate(epoch_idxs):
+            this_epochs = self[epoch_idx]
```

Comment:
this only issue I see here is that it will make a copy of epochs even if there is only one part. The way out is to check if epoch_idx contains all indices which boils down to checking that len(epoch_idxs) == 1

@mainakjas let's just avoid that.